dm/config/subtask.go

```diff
@@ -338,6 +344,54 @@ func (c *SubTaskConfig) Adjust(verifyDecryptPassword bool) error {
 		log.L().Warn("'online-ddl-scheme' will be deprecated soon. Recommend that use online-ddl instead of online-ddl-scheme.")
 	}
 
+	if len(c.ShadowTableRule) == 0 {
+		c.ShadowTableRule = []string{"^_(.+)_(?:new|gho)$"}
+	} else {
+		var shadowTableRule []string
+		for _, r := range c.ShadowTableRule {
+			if !strings.HasPrefix(r, "^") {
+				r = "^" + r
+			}
+
+			if !strings.HasSuffix(r, "$") {
+				r = r + "$"
+			}
+
+			p, err := regexp.Compile(r)
+			if err != nil {
+				return err
+			}
+			if p.NumSubexp() != 1 {
+				return terror.Annotate(err, "'shadow-table-rule' regex pattern isn't contains exactly one submatch")
```

Comment:
`err` is actually `nil` here. We should add a new `terror` and print the `ShadowTableRule` here.
Example:
```
				return terror.ErrConfigOnlineDDLInvalidRegex.Generate("'shadow-table-rule", r)
```
Defenition in [terror](https://github.com/pingcap/dm/blob/master/pkg/terror/error_list.go):
```
	ErrConfigOnlineDDLInvalidRegex   = New(codeConfigOnlineDDLInvalidRegex, ClassConfig, ScopeInternal, LevelHigh, "''%s' regex pattern %s isn't contains exactly one submatch")
```